api_3/users_get_common.py

```diff
@@ -66,16 +66,31 @@ def get(self, program):
                 das = DatasetAccessSupportFactory.from_webapp_django_settings()
                 authorized_datasets = das.get_datasets_for_era_login(user_email)
                 for dataset in authorized_datasets:
-                    ad = AuthorizedDataset.objects.filter(whitelist_id=dataset.dataset_id)
-                    if program in ad.name:
-                        authorized = True
-                        allowed = True
+                    try:
+                        ad = AuthorizedDataset.objects.get(whitelist_id=dataset.dataset_id)
+                        if program in ad.name:
+                            authorized = True
+                    except (ObjectDoesNotExist) as e:
+                        logger.exception('didn\'t find an expected authorized dataset for {}: {}'.format(dataset, e))
+                        raise
+                    except (MultipleObjectsReturned) as e:
+                        authdatasets = AuthorizedDataset.objects.filter(whitelist_id=dataset.dataset_id).values_list('name',flat=True)
+                        logger.exception('found more than one expected authorized dataset for {} named {}: {}'.format(dataset, ", ".join(authdatasets), e))
+                        raise
                 if not allowed:
                     user_auth_datasets = AuthorizedDataset.objects.filter(id__in=UserAuthorizedDatasets.objects.filter(nih_user_id=nih_user.id).values_list('authorized_dataset', flat=True))
                     for dataset in user_auth_datasets:
-                        ad = AuthorizedDataset.objects.filter(whitelist_id=dataset.dataset_id)
-                        if program in ad.name:
-                            allowed = True
```

Comment:
Quick note, in this section you don't need to do the try/except and the get because you don't need to fetch the dataset again. A given dataset from user_auth_datasets is an instance of an AuthorizedDataset, since you used filter and explicitly went through the Django ORM to get them. They won't show up if a user isn't authorized, which is fine--that's what you're trying to find out, so a 0 answer is acceptable, and you don't need to double-check that there aren't multiple entries again. (Feel free to drop by my desk if this doesn't make sense.)